### PR TITLE
hash typed data

### PR DESCRIPTION
**Pull request by gravityblast**
Added `HashTypedData` so that this hash can be signed with a keycard.

### CODE REVIEW

File: services/typeddata/hash.go [hunk 1]
```diff
@@ -19,6 +19,15 @@ var (
 	int256Type, _  = abi.NewType("int256", nil)
 )
 
+// Hash generates a hash of TypedData and verifies that chainId in the typed data matches currently selected chain.
+func Hash(typed TypedData, chain *big.Int) (common.Hash, error) {
```
Comment (dshulyak) [inline services/typeddata/hash.go]: i would rename to something more verbose cause this method also performs some validation, and hashing usually doesn't

**Review by dshulyak — APPROVED** [review]

File: services/typeddata/hash.go (hunk 1 above)
Comment (gravityblast) [reply services/typeddata/hash.go]: @dshulyak good idea, something like `ValidateAndHash`?

Comment (dshulyak) [reply services/typeddata/hash.go]: sounds fine :+1:

**Comment by jakubgs** [comment]
I've merged #1427, please rebase this.

**Pull request merged**